app/controllers/concerns/course_basic_detail_concern.rb

```diff
@@ -145,27 +143,43 @@ def build_meta_course_creation_params
     @meta_course_creation_params = params.slice(:goto_confirmation)
   end
 
+  def continue_step
+    params[:goto_confirmation] && !%i[subjects].include?(current_step) ? :confirmation : CourseCreationStepService.new.execute(current_step: current_step, course: @course)[:next]
```

Comment:
wat is this doing?

I'd break this up for the sake of legibility. For example, why `!%i[subjects].include?(current_step)` and not `current_step != :subject`, is that not the same thing?

I think it'd also benefit from being broken up onto multiple lines.